src/js/hopscotch.js

```diff
@@ -21,6 +21,7 @@
       hasSessionStorage = false,
       isStorageWritable = false,
       document          = window.document,
+      validIdRegEx      = /^[a-zA-Z0-9_-]+$/,
```

Comment:
This regex allows that IDs that we won't be able to use as class names. For simplicity sake, let's enforce ID to start with alpha character which can be followed by  0 or more alhpanumeric characters, hyphen or underscore. Something like this `/^[a-zA-Z]+[a-zA-Z0-9_-]*$/`